app/Http/Requests/EntryUpdateRequest.php

```diff
@@ -49,7 +50,7 @@ public function rules()
             'gp' => ['sometimes', 'numeric', 'between:-999999999999999999999999999999.99,999999999999999999999999999999.99'],
             'downtime' => ['sometimes', 'integer'],
             'notes' => ['nullable', 'string'],
-            'items' => ['sometimes', 'array'],
+            'items' => ['sometimes', 'array', $requiredChoice],
```

Comment:
Hey @m-triassi, could you explain the `$requiredChoice` expression? The way I read it is that an array of items is required if the user's chose `Magic Items` as a reward and the items in the request were empty. Shouldn't the `items` array not be empty?

Reply:
What you're saying is mostly correct, it just stipulates that there should be _something_ for magic items if the choice is "magic_items", in the `$requiredChoice` we check both what the value of choice is **and** that the items array isn't empty. 

So to summerize, the `$requiredChoice` is --> is choice == "magic_item"? yes. is `items` not empty? yes --> you're good, if not then validation fails.